clients/admin-ui/src/features/common/custom-fields/CustomFieldsList.tsx

```diff
@@ -0,0 +1,224 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Center, Divider, Flex, Spinner, Text } from "@fidesui/react";
+import { FieldArray, useFormikContext } from "formik";
+import { useCallback, useEffect, useMemo, useState } from "react";
+
+import { CustomMultiSelect, CustomSelect } from "~/features/common/form/inputs";
+import { useAlert } from "~/features/common/hooks";
+import {
+  useGetAllAllowListQuery,
+  useGetCustomFieldDefinitionsByResourceTypeQuery,
+  useGetCustomFieldsForResourceQuery,
+  useUpsertCustomFieldMutation,
+} from "~/features/plus/plus.slice";
+import { AllowedTypes, ResourceTypes } from "~/types/api";
+
+import { useFeatures } from "../features";
+import { CustomFieldsModal } from "./CustomFieldsModal";
+import { CustomFieldWithIdExtended } from "./types";
+
+const initialValues = {
+  customFields: [] as CustomFieldWithIdExtended[],
+};
+
+type FormValues = typeof initialValues;
+
+type CustomFieldsListProps = {
+  resourceId: string;
+  resourceType: ResourceTypes;
+};
+
+const CustomFieldsList = ({
+  resourceId,
+  resourceType,
+}: CustomFieldsListProps) => {
+  const { errorAlert, successAlert } = useAlert();
+
+  const { flags, plus } = useFeatures();
+  const skip = !(flags.customFields && plus);
+  const [isLoading, setIsLoading] = useState(!skip);
```

Comment:
We should move away from storing `isLoading` in state. When I opened this page the API returned an error, but the spinner stuck around even though the app wasn't waiting for anything.

`isLoading` can always be derived some other state, such as the loading state of multiple queries:
https://github.com/ethyca/fides/blob/c0866f3f5f8c68c7eaaae0fc7c3ce294397bf095/clients/admin-ui/src/features/dataset/DatasetCollectionView.tsx#L53

Additional logic like the `useEffect` below could be a `useMemo` instead, but usually it's just boolean operations that can done inline without worrying about render optimization (and actually introducing memo checks introduces _more_ comparison operations).